src/components/_global/BalForm/readme.md

```diff
@@ -0,0 +1,99 @@
+# BalForm
+
+`<BalForm />` is a useful component for creating wrappers for simple and complex form structures on our frontend. There are a few motivations behind using a form component for creating forms:
+
+- Centralised and instance specific state. The form owns its state and can pass down its state (aka form values) to children at any nested level. This also means not needing to creating and managing your own form state in whichever component / composable that might need to feed into your form.
+- Centralised validation. If your component accepts in a rules prop, you can integrate it as a form component (more on this below) and have it validate against all passed in rules, whenever the components value changes. Any errors present during submission of the form will prevent the submission function from executing and a list of errors for each field be exported by the forms instance. 
+- Simplified integration**. All you need to do is define the form component you are using as a child of `<BalForm />` to have its values, rules and any change handlers bound to the form, automatically.
+
+  ** Assuming that all child form components (e.g. inputs, radio buttons, complex inputs such as <TokenInput/> have code that allows them to integrate as a form component.
+
+
+## Usage
+
+```typescript
+// somecomponent.vue
+
+/// script
+
+// defines a component instance that will be
+// passed into <BalForm />. Names are tied to 
+// form instances and are unique. The value returned
+// represents the form instance.
+const form = useForm({ name: 'my-form' })
+
+function mySubmit(formValues: T) {
+ // do stuff
+}
+
+/// template
+<BalForm :form="form" :submit="form.handleSubmit(mySubmit)"
+  <div v-if="form.errors.username.length > 0">There are errors in your username</div>
+  <TokenInput name="tokenIn" />
+  <input :value="form.register('username', [isLowerCase()])" />
+</BalForm>
+```
+
+In the above example. `<TokenInput />` is registered as a complex form component. However, all that needs to be done to register this component to the forms state is to simply give it a name and have it as a child of `<BalForm />`. 
```

Comment:
I'd use the terminology 'wrapped input' here rather than 'complex component'
```suggestion
In the above example. `<TokenInput />` is registered as a wrapped form input. However, all that needs to be done to register this component to the forms state is to simply give it a name and have it as a child of `<BalForm />`. 
```